### PR TITLE
README improvements

### PR DESCRIPTION
**Pull request by brandjon**
Follow-up to #237. Moves development workflow info to CONTRIBUTING.md. Fixes
the instructions for how to use the Bazel Federation.

Also updates the project WORKSPACE to use a released version of the Federation.

### CODE REVIEW

**Comment by brandjon** [comment]
Ping @laurentlb

**Comment by brandjon** [comment]
Adding @fweikert since aiuto is on vacation. Florian, this is just a readme update that happens to include some fixes for the federation, so I figured you could vet the federation content ("Getting started" section) and rubber stamp the rest.

File: README.md [hunk 1]
```diff
@@ -31,21 +31,44 @@ The packaging rules (`pip_import`, etc.) are less stable. We may make breaking
...
+Federation](https://github.com/bazelbuild/bazel-federation), you just need to
+[import the Federation](https://github.com/bazelbuild/bazel-federation#example-workspace)
+and call the rules_python setup methods:
 
 ```python
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+
+http_archive(
+    name = "bazel_federation",
+    url = "https://github.com/bazelbuild/bazel_federation/releases/download/0.0.1/bazel_federation-0.0.1.tar.gz",
```
Comment (fweikert) [inline README.md]: Incorrect URL: the repo is bazel-federation, not bazel_federation (but the underscore in the basename is actually correct).

I noticed that the federation's release notes contain the same error :(

Comment (brandjon) [reply README.md]: This is really, really confusing. When I copy paste that URL into my browser, it serves me a page saying "Not found". When I edit the underscore in the repo name (not the artifact) into a hyphen, it downloads the artifact but redirects me to the original URL with the incorrect underscore.

Comment (brandjon) [reply README.md]: Pushed the 1-character change.

Comment (fweikert) [reply README.md]: I had the same thing - apparently Chrome replaces the address of a successfully downloaded file with the previous one.

Comment (brandjon) [reply README.md]: Ok, glad it's not a bad github redirect.

**Comment by brandjon** [comment]
I'll merge this. Laurent, feel free to request changes in a follow-up.

**Comment by laurentlb** [comment]
Thanks!
(sorry for the delay)

**Pull request merged**